File: AtCoder/394B-cat.rs
use std::io;
use std::str::FromStr;

struct Scanner {
    buffer: Vec<String>,
}

impl Scanner {
    fn new() -> Scanner {
        Scanner { buffer: Vec::new() }
    }

    fn next<T: FromStr>(&mut self) -> T {
        loop {
            if let Some(token) = self.buffer.pop() {
                return token.parse().ok().expect("Failed parse");
            }
            let mut input = String::new();
            io::stdin().read_line(&mut input).expect("Failed read");
            self.buffer = input.split_whitespace().rev().map(String::from).collect();
        }
    }
}

fn solve(vec: &mut Vec<String>, n: usize) -> String {
    for i in 0..n {
        let mut min = i;
         for j in (i + 1)..n {
             if vec[j].len() < vec[min].len() {
                 min = j;
             }
         }
         if min != i { vec.swap(i, min); }
    }

    let mut ans = String::new();
    for str in vec {
        ans += str;
    }
    return ans;
}

fn main() {
    let mut scan = Scanner::new();

    let n: usize = scan.next();
    let mut vec: Vec<String> = (0..n).map(|_| scan.next::<String>()).collect();

    let ans = solve(&mut vec, n);
    println!("{}", ans);
}

File: AtCoder/394B-cat.cpp
#include <iostream>
#include <string>
#include <vector>

std::string solve(std::vector<std::string>& s, int n) {
  for (int i = 0; i < n; i++) {
    int min = i;
    for (int j = i + 1; j < n; j++) {
      if (s[j].length() < s[min].length()) min = j;
    }
    if (min != i) std::swap(s[i], s[min]);
  }

  std::string ans = "";
  for (auto x : s) {
    ans += x;
  }
  return ans;
}

int main() {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(nullptr);
  
  int n;
  std::cin >> n;

  std::vector<std::string> s(n);
  for (auto& x : s) 
    std::cin >> x;

  std::string ans = solve(s, n);
  std::cout << ans;
}
